src/garage/torch/algos/td3.py

```diff
@@ -0,0 +1,429 @@
+        self._target_qf_2 = copy.deepcopy(self._qf_2)
+
+        self._policy_optimizer = make_optimizer(policy_optimizer,
+                                                module=self.policy,
+                                                lr=policy_lr)
+        self._qf_optimizer_1 = make_optimizer(qf_optimizer,
+                                              module=self._qf_1,
+                                              lr=qf_lr)
+        self._qf_optimizer_2 = make_optimizer(qf_optimizer,
+                                              module=self._qf_2,
+                                              lr=qf_lr)
+        self._actor_loss = torch.zeros(1)
+
+    def __getstate__(self):
+        """Object.__getstate__.
+
+        Returns:
+            dict: the state to be pickled for the instance.
+
+        """
+        data = self.__dict__.copy()
+        del data['_replay_buffer']
+        return data
+
+    def __setstate__(self, state):
+        """Object.__setstate__.
+
+        Args:
+            state (dict): unpickled state.
+
+        """
+        self.__dict__.update(state)
+        self._replay_buffers = PathBuffer(
+            capacity_in_transitions=self._replay_buffer_size)
+
+    def _get_action(self, action, noise_scale):
+        """Select action based on policy.
+
+        Action can be added with noise.
+
+        Args:
+            action (float): Action.
+            noise_scale (float): Noise scale added to action.
+
+        Return:
+            float: Action selected by the policy.
+        """
+        action += noise_scale * np.random.randn(self._action_dim)
+        return np.clip(action, -self._max_action, self._max_action)
+
+    def train(self, trainer):
+        """Obtain samplers and start actual training for each epoch.
+
+        Args:
+            trainer (Trainer): Experiment trainer, which provides services
+                such as snapshotting and sampler control.
+
+        Returns:
+            float: The average return in last epoch cycle.
+        """
+        if not self._eval_env:
+            self._eval_env = trainer.get_env_copy()
+        last_returns = None
+        trainer.enable_logging = False
+        for _ in trainer.step_epochs():
+            for cycle in range(self._steps_per_epoch):
+                # Obtain trasnsition batch and store it in replay buffer.
+                # Get action randomly from environment within warm-up steps.
+                # Afterwards, get action from policy.
+                if self._uniform_random_policy and \
+                        trainer.step_itr < self._start_steps:
+                    trainer.step_path = trainer.obtain_episodes(
+                        trainer.step_itr,
+                        agent_update=self._uniform_random_policy)
+                else:
+                    trainer.step_path = trainer.obtain_episodes(
+                        trainer.step_itr, agent_update=self.exploration_policy)
```

Comment:
It's pretty confusing that `self.exploration_policy` can be `None`, in which case this call actually causes `self.policy` to be used. Can you make the logic here explicit so we don't accidentally break it when we refactor?